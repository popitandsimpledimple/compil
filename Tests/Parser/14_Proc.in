procedure abc(var a: Double; const b: Integer; c: String);
begin
end;
procedure asd();
begin
end;

procedure procedure2();
var z: Integer; x: Integer = 123;
begin
    writeln('Hello, world');
end;


begin
end.