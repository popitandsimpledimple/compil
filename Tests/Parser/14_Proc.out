program
   decls
      procedure
         abc
      params
         var
            a
            type
               double
         const
            b
            type
               integer
         param
            c
            type
               string
         block
            compounds statements
      procedure
         asd
      params
         block
            compounds statements
      procedure
         procedure2
      params
         decls
            vars
               var
                  z
                  type
                     integer
                  no expression
               var
                  x
                  type
                     integer
                     123
         block
            compounds statements
               call
                  writeln
                     'hello, world'
   block
      compounds statements
